web/src/components/upload/UploadsTableDeleteDialog.vue

```diff
@@ -31,14 +33,10 @@ const confirm = async (): Promise<void> => {
 
   try {
     // Attempt to delete the upload.
-    await axios.delete(
-      `${process.env.VUE_APP_API_URL}/reports/${props.report.id}`
-    );
+    await axios.delete(appmode.reportUrl.value);
```

Comment:
If uploading a dataset works, but creating a report fails (because of an error), that report will have a local `error` status without associated report ID. Trying to delete this report here will however also fail (because the id is `undefined` and the report does not exist on the server).

I think if there is no associated report id, we can just delete it locally.

In addition, maybe we should automatically delete failed reports if they are older than a day?